core/src/com/unciv/logic/city/CityStats.kt

```diff
@@ -333,9 +333,11 @@ class CityStats(val city: City) {
         if (!city.civ.isHuman()) {
             buildingsMaintenance *= city.civ.gameInfo.getDifficulty().aiBuildingMaintenanceModifier
         }
-
+        
         for (unique in city.getMatchingUniques(UniqueType.BuildingMaintenance)) {
-            buildingsMaintenance *= unique.params[0].toPercent()
+            city.cityConstructions.getBuiltBuildings().filter { it.matchesFilter(unique.params[1]) }.forEach { 
```

Comment:
No, this is wrong - you're altering the total maintenance, when it's only the specific building which should be affected
Sounds to me like this should be in CityConstructions.getMaintenanceCosts